backend/src/main/java/services/UserCriteriaSearch.java

```diff
@@ -0,0 +1,51 @@
+package services;
+import data.entities.UserEntity;
+import data.entities.UserEntity_;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.CriteriaQuery;
+import jakarta.persistence.criteria.Predicate;
+import jakarta.persistence.criteria.Root;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@ApplicationScoped
+public class UserCriteriaSearch {
+
+    private final EntityManager entityManager;
+
+    @Inject
+    public UserCriteriaSearch(EntityManager entityManager) {
+        this.entityManager = entityManager;
+    }
+
+    public List<UserEntity> searchUsers(String userId, String firstName, String lastName, String email, String plant) {
+
+        CriteriaBuilder cb = entityManager.getCriteriaBuilder();
+        CriteriaQuery<UserEntity> query = cb.createQuery(UserEntity.class);
+        Root<UserEntity> root = query.from(UserEntity.class);
+        List<Predicate> predicates = new ArrayList<>();
+
+        if (userId != null && !userId.isEmpty()) {
```

Comment:
Just hint for future: there is StringUtils.isEmpty and StringUtils.isNotEmpty for these checks